src/rapids_dependency_file_generator/cli.py

```diff
@@ -87,18 +115,29 @@ def main(argv=None):
     validate_dependencies(parsed_config)
 
     matrix = generate_matrix(args.matrix)
-    to_stdout = all([args.file_key, args.output, args.matrix is not None])
 
-    if to_stdout:
-        includes = parsed_config["files"][args.file_key]["includes"]
+    if len(args.file_key) > 0:
         parsed_config["files"] = {
-            args.file_key: {
-                "matrix": matrix,
-                "output": args.output,
-                "includes": includes,
-            }
+            file_key: parsed_config["files"][file_key]
+            for file_key in args.file_key
+            if parsed_config["files"].get(file_key, None) is not None
         }
 
+    if len(args.output) > 0:
+        files = {}
+        for file_key, file_config in parsed_config["files"].items():
+            outputs = get_requested_output_types(file_config["output"])
+            outputs = list(filter(lambda output: output in outputs, args.output))
+            if len(outputs) > 0:
+                file_config["output"] = outputs
```

Comment:
This PR supports passing multiple output types, so the logic as-is is correct. I filter the outputs for ones that occur in `args.output` on L130.

Reply:
I got confused by the logic swapping which variable means what here because the filtered `args.output` is being assigned back to `outputs` after checking for membership in `outputs`. You are right, the filtering makes sense.